### PR TITLE
Fix: go tests for pipelineLoop controller.

### LINKED ISSUE #644: pipelineloop unit tests are broken after Tekton 0.25 upgrade
/kind bug

**What steps did you take and what happened:**
when running `go test -v -cover ./tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/`. Several unit tests are broken for pipelineloop, so we need to fix those tests.

**What did you expect to happen:**

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

* Python Version (use `python --version`):
* SDK Version: 
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by ScrapCodes**
**Which issue is resolved by this Pull Request:** 

#644

**Description of your changes:**
Fixed tests,

And for those tests, that are not yet fixed. ADDED a TODO to do it later.

Following failures:

```
  pipelinelooprun_test.go:1261: Error reconciling: 1 error occurred:
177
        	* error creating PipelineRun from Run run-pipelineloop: pipelineruns.tekton.dev "run-pipelineloop-00001-9l9zj" already exists
```
**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by Tomcli** [comment]
looks like the test timed out, could be something get stuck during the test.
/cc @pugangxa

**Comment by ScrapCodes** [comment]
Some of the failing tests are removed.

**Comment by ScrapCodes** [comment]
@Tomcli doc link is fixed - hopefully.

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go [hunk 1]
```diff
@@ -883,40 +847,6 @@ var expectedPipelineRunIteration1 = &v1beta1.PipelineRun{
 	},
 }
 
-var expectedConditionPipelineRunIteration1 = &v1beta1.PipelineRun{
```
Comment (Tomcli) [inline tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go:886]: can you add some TODO comments for adding some condition pipeline unit tests? It's broken right now but we want to bring it back later on.

Comment (ScrapCodes) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go:886]: Yes, I can. Alternatively, leave the issue open. If I get another opportunity, I will work on it.

Comment (Tomcli) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go:886]: thanks

Comment (ScrapCodes) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go:886]: Added a TODO. If you like, In the mean time, you can merge this PR

**Pull request merged**